src/Protocol/Diaspora.php

```diff
@@ -388,7 +389,11 @@ public static function decodeRaw(array $importer, $raw)
 
 			if (!is_object($j_outer_key_bundle)) {
 				logger('Outer Salmon did not verify. Discarding.');
-				System::httpExit(400);
+				if ($no_exit) {
+					return false;
+				} else {
+					System::httpExit(400);
```

Comment:
This should be done in the calling module instead of here. You got the right idea adding a `return false;`, but I think you didn't bring it far enough. There should be no flag in the function parameter, and `System::httpExit()` should be called in the calling modules.

Reply:
So close before a release I hadn't intended to change too much.

Reply:
I created a tracking issue for the next release.